nevergrad/parametrization/core.py

```diff
@@ -276,9 +276,17 @@ def satisfies_constraints(self) -> bool:
         if not self._constraint_checkers:
             return True
         val = self.value
-        return all(func(val) for func in self._constraint_checkers)
+        def ok(x: tp.Union[bool, float]) -> bool:
+            return x >= 0. if isinstance(x, float) else x
+        return all(ok(func(val)) for func in self._constraint_checkers)
 
-    def register_cheap_constraint(self, func: tp.Callable[[tp.Any], bool]) -> None:
+    def penalty(self, num_ask: int, budget: int, exponent: float):
+        def float_penalty(x: tp.Union[bool, float]) -> float:
+            return 1 if x is False else 0 if x is True else -x if x < 0 else 0
+        val = self.value
+        return (exponent ** (num_ask / np.sqrt(budget))) * sum(float_penalty(func(val)) for func in self._constraint_checkers)
```

Comment:
this is way too linked to the optimization to be here, let's move this in the optimizer

Reply:
Done.